File: amazon/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& A) {
        
        int arrSize = (int)A.size();
        
        vector<int> result(arrSize);
        int l = 0;
        int r = arrSize - 1;
        int i = arrSize - 1;
                
        while (l <= r) {
            
            int l_sqr = A[l] * A[l];
            int r_sqr = A[r] * A[r];
            
            if (l_sqr >= r_sqr) {                
                result[i] = l_sqr;
                l++;
            }
            else {
                result[i] = r_sqr;
                r--;
            }
            i--;
        }
        
        return result;
    }
};

File: amazon/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        
        left = 0
        right = len(A) - 1
        pos = len(A) - 1
        
        # empty list
        res = [None] * len(A)
        
        # add elements in decreasing order
        # starting from largest to smallest
        while left <= right:
            
            x = A[left]**2
            y = A[right]**2
            
            if x >= y:
                res[pos] = x
                left += 1
            else:
                res[pos] = y
                right -= 1
                
            pos -= 1
            
        return res
